src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackagesErrorBar.xaml.cs

```diff
@@ -0,0 +1,125 @@
+
+        public event EventHandler RefreshUI;
+
+        public event EventHandler InitializationCompleted;
+
+        public PackageExtractionContext PackageExtractionContext { get; set; }
+
+        public ISourceControlManagerProvider SourceControlManagerProvider { get; }
+
+        public ProjectManagement.ExecutionContext ExecutionContext { get; }
+
+        public XDocument OriginalPackagesConfig { get; set; }
+
+        public NuGetActionType ActionType { get; set; }
+
+        public TelemetryServiceHelper TelemetryService { get; set; }
+
+        public PackagesErrorBar(ISolutionManager solutionManager)
+        {
+            InitializeComponent();
+
+            _solutionManager = solutionManager;
+
+            // Start a task to check for all the project's packages file and accordingly update error bar.
+            // this would also be initlize VSSolutionManager first time, if it hasn't been initialized yet.
+            Task.Run(UpdateErrorBarAsync);
+
+            ErrorMessage.SetResourceReference(TextBlock.ForegroundProperty, VsBrushes.InfoTextKey);
+            ErrorBar.SetResourceReference(Border.BackgroundProperty, VsBrushes.InfoBackgroundKey);
+            ErrorBar.SetResourceReference(Border.BorderBrushProperty, VsBrushes.ActiveBorderKey);
+        }
+
+        private async Task UpdateErrorBarAsync()
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            try
+            {
+                var projects = (await _solutionManager.GetNuGetProjectsAsync()).ToArray();
+                foreach (var project in projects)
+                {
+                    await project.GetInstalledPackagesAsync(CancellationToken.None);
+                }
+
+                ErrorBar.Visibility = Visibility.Collapsed;
+            }
+            catch (Exception ex)
+            {
+                ErrorMessage.Text = ex.Message;
+                ErrorBar.Visibility = Visibility.Visible;
+            }
+            finally
+            {
+                InitializationCompleted?.Invoke(this, EventArgs.Empty);
+            }
+        }
+
+        private void ExecuteRefresh(object sender, EventArgs e)
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(UpdateErrorBarAsync);
+
+            // raise RefreshUI event to refresh the manager ui
+            RefreshUI?.Invoke(this, EventArgs.Empty);
+        }
+
+        public void Log(MessageLevel level, string message, params object[] args)
+        {
+            if (args.Length > 0)
+            {
+                message = string.Format(CultureInfo.CurrentCulture, message, args);
+            }
+
+            ShowMessage(message);
+        }
+
+        public void CleanUp()
+        {
+
+        }
+
+        public void ReportError(string message)
+        {
+            ShowMessage(message);
+        }
+
+        public FileConflictAction ResolveFileConflict(string message)
```

Comment:
I don't mind that, but it's implicit when you implement this interface. And this API isn't even called explicitly in the current context.